gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    """
+    if scipy.sparse.issparse(vec):
+        vec = vec.todense().tolist()
+    try:
+        id_, val_ = vec[0]
```

Comment:
In this use-case of similarity functions an empty input does not make any sense, but yes, I'll handle that case. What would be the ideal thing to do if it is indeed empty?